substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/thread/Target_java_lang_Thread.java

```diff
@@ -458,6 +458,13 @@ public static boolean interrupted() {
      */
     @Substitute
     void interrupt0() {
+        Thread thread = JavaThreads.fromTarget(this);
+
+        // Interrupting a thread that is not alive need not have any effect.
+        if (!PlatformThreads.isAlive(thread)) {
```

Comment:
Did you see any particular issue without this check?

The check that you added is racy and it is also something that HotSpot does not do (i.e., HotSpot doesn't look at the Java thread state).